src/app/ad-authoring/ad-authoring.state.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import {Observable, BehaviorSubject} from 'rxjs';
+import {AdAuthoring} from './ad-authoring';
+
+export interface AdAuthoringState {
+  readonly adAuthorings: ReadonlyArray<AdAuthoring>;
```

Comment:
I'm not sure we need an array here, the user will be working on one Ad at a time.